### PR TITLE
Add wait_for_topic option to Kafka

### PR DESCRIPTION
**Pull request by agrare**
Add an option to Kafka subscribe_messages and subscribe_topic to wait for topic creation with increasing backoff retry.

### CODE REVIEW

File: lib/manageiq/messaging/kafka/queue.rb [hunk 1]
```diff
@@ -17,13 +17,18 @@ def publish_messages_impl(messages)
...
+          wait_for_topic = true if wait_for_topic.nil?
+
           topic = address(options)
           options[:persist_ref] = GROUP_FOR_QUEUE_MESSAGES + topic
 
-          queue_consumer = consumer(true, options)
-          queue_consumer.subscribe(topic)
-          queue_consumer.each do |message|
-            process_queue_message(queue_consumer, topic, message, &block)
+          wait_for_topic(wait_for_topic) do
```
Comment (Fryguy) [inline lib/manageiq/messaging/kafka/queue.rb:26]: I'm amazed this works 😆

File: lib/manageiq/messaging/kafka/common.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def consumer(beginning, options)
 
         def raw_publish(body, options)
           options[:payload] = encode_body(options[:headers], body)
-          producer.produce(**options)
+          producer.produce(options)
```
Comment (Fryguy) [inline lib/manageiq/messaging/kafka/common.rb:26]: Is this a bug that was fixed?

Comment (agrare) [reply lib/manageiq/messaging/kafka/common.rb:26]: Oh yes, missed this when resolving a conflict

**Comment by agrare** [comment]
Testing with MIQ, passing false fails immediately:
```
>> MiqQueue.messaging_client("test").subscribe_messages(:service => "abcd", :wait_for_topic => false) { |msg| puts msg.inspect }
/home/grare/adam/.gem/ruby/3.0.0/gems/rdkafka-0.12.0/lib/rdkafka/consumer.rb:432:in `poll': Broker: Unknown topic or partition (unknown_topic_or_part) (Rdkafka::RdkafkaError)
	from /home/grare/adam/.gem/ruby/3.0.0/gems/rdkafka-0.12.0/lib/rdkafka/consumer.rb:458:in `block in each'
```

Passing true blocks until a message is published:
```
>> MiqQueue.messaging_client("test").subscribe_messages(:service => "abcd", :wait_for_topic => true) { |msg| puts msg.inspect }

...

(separate console `> MiqQueue.messaging_client("test").publish_message(:service => "abcd", :messag
e => "hello", :payload => {"hello" => "world"})` 

[#<ManageIQ::Messaging::ReceivedMessage:0x0000563583399478 @sender=nil, @message="hello", @payload={"hello"=>"world"}
```

**Pull request merged**